framework/include/controls/ControlInterface.h

```diff
@@ -228,6 +146,18 @@ ControlInterface::getValidInputParameterIteratorsByName(const std::string & name
   if (output.size() == 0)
     mooseError("The controlled parameter, '" << name << "', in " << _ci_name << " was not found.");
 
+  // Error if the size limit was reached
+  if (count != std::numeric_limits<unsigned int>::max() && output.size() != count)
```

Comment:
I could change it to 99999 :smile:  

I just wanted to provide a method for producing some sort of an error if you are trying to control things that are expected to be controlled. I am open for other ways.